org.uqbar.project.wollok.ui.diagrams/src/org/uqbar/project/wollok/ui/diagrams/AbstractDiagramView.xtend

```diff
@@ -161,7 +164,13 @@ abstract class AbstractDiagramView extends ViewPart implements ISelectionListene
 					ZoomManager.FIT_HEIGHT
 				])
 				zoomIn = new ZoomInAction(zoomManager)
+				if (environmentHasDarkTheme) {
+					zoomIn.imageDescriptor = ImageDescriptor.createFromFile(this.class, "/icons/zoomIn-dark.png") 
+				}
 				zoomOut = new ZoomOutAction(zoomManager)
+				if (environmentHasDarkTheme) {
+					zoomOut.imageDescriptor = ImageDescriptor.createFromFile(this.class, "/icons/zoomOut-dark.png") 
+				}
```

Comment:
Estos ifs terminan estando en todos los lugares donde algo cambie respecto al tema. Es muy costoso delegar a algún objeto donde se le pida los iconos / colores en función del tema? Sospecho que se podría aprovechar desde varios lugares.

Reply:
lo que más logré hacer en algunos lados fue tener un _extension method_ que se llama themeSuffix() que justamente le concatena el `-dark` al ícono. Pero ojo, acá no puedo hacer eso porque el image descriptor sale del zoom in/zoom out que es un ícono propio de Eclipse. Es decir, un ícono sale del directorio _icons_ y otro del platform:plugin. Las formas de obtenerlos son distintos.

En resumen, Eclipse hizo un refurbishing de la forma en que se visualizan los themes a partir de Mars o Neon, creo, no estoy seguro. Ahí empezaron a usar css, por eso te funciona hacer Ctrl + `+` para hacer zoom sobre el editor. El Dark Mode vino un par de versiones después, y la verdad es que no hay una documentación simple de cómo definir un tema con el css, sí vas a conseguir algo como [este artículo](https://www.vogella.com/tutorials/Eclipse4CSS/article.html), pero no es fácil enganchar justo cómo se llama el control que querés estilear. Si encima son los de Wollok, es casi imposible, porque fijate que

- el diagrama estático
- el dinámico
- la consola

definen colores a manopla. E incluso la consola y el editor lo definen de dos maneras distintas. Y el outline necesita íconos que se vean bien con fondo negro y blanco, pero los íconos son de 16x16, es muy muy muy difícil lograr que se entienda bien qué es lo que refleja. Visual Studio Code en ese sentido está más preparado, mostrando menos info pero más grande, y se adapta mejor.

En definitiva, si le podés encontrar una vuelta yo te sigo, a mí mucho más que el suffixTheme() no se me ocurrió.
Abracín!